src/activateTelemetry.ts

```diff
@@ -0,0 +1,88 @@
+import * as vscode from "vscode";
+import { PostHog } from "posthog-node";
+import { randomBytes } from "crypto";
+
+export type TrackEvent = (
+  eventName: string,
+  properties?: { [key: string]: any }
+) => void;
+
+const key = "phc_pCt2zPQylp5x5dEKMB3TLM2hKBp7aLajUBgAfysPnpd";
+const host = "https://eu.i.posthog.com";
+
+let posthogClient: PostHog | undefined;
+let trackingAllowed = false;
+
+export function activateTelemetry(
+  context: vscode.ExtensionContext
+): TrackEvent {
+  const config = vscode.workspace.getConfiguration("telemetry");
+  trackingAllowed = config.get<boolean>("enableTelemetry", true);
+  const vscodeVersion = vscode.version;
+  const osPlatform = process.platform;
+  const extension = vscode.extensions.getExtension("nextflow.nextflow");
+  const extensionVersion = extension?.packageJSON.version ?? "unknown";
+
+  if (!trackingAllowed) return () => {};
+
+  const trackEvent = createTrackEvent(context);
+
+  posthogClient = new PostHog(key, { host });
+
+  trackEvent("extensionActivated", {
+    extensionVersion,
+    vscodeVersion,
+    osPlatform,
+  });
+
+  // Track file open events
+  // TODO: check privacy rules for vscode extensions (here we're storing the user's full file path)
```

Comment:
to your point we might just be able to put this in the readme or have the user give us consent via terms.